### PR TITLE
Adds new end-points and improves to existing ones

### PR DESCRIPTION
**Pull request by sk-sahu**
## New API end-points 
* Create a Cohort
* Extract Genotypic Data

Also improvements to existing ones with status check `200` for API calls and error message.

### CODE REVIEW

File: .github/workflows/R-CMD-check.yaml [hunk 1]
```diff
@@ -1,12 +1,5 @@
-on:
-  push:
-    branches:
-      - master
-  pull_request:
-    branches:
-      - master
-
 name: R-CMD-check
+on: [push, pull_request]
```
Comment (cgpu) [inline .github/workflows/R-CMD-check.yaml:2]: check all the things sounds good to me!

File: R/extract_genotypic_data.R [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+#' @param page_size Number of entries in a page. (Optional) Default - 10
+#' @param filters WIP - details will be added.
+#'
+#' @return A dataframe.
+#'
+#' @examples
+#' \dontrun{
+#' extract_genotypic_data(baseurl= "https://cloudos.lifebit.ai",
+#'              auth = "Bearer ***token***",
+#'              teamid = "***teamid***")
```
Comment (cgpu) [inline R/extract_genotypic_data.R:19]: can you make everythang snakecase eg
`baseurl` -> `base_url`
`teamid ` -> `team_id`

in all occurrences?

File: R/extract_participants.R [hunk 3]
```diff
@@ -25,7 +25,11 @@ extract_participants <- function(baseurl, auth, raw_data) {
...
                   encode = "json"
         )
-  res <- httr::content(r, as = "text")
-  df <- utils::read.csv(textConnection(res))
-  return(df)
+  if (!r$status_code == 200) {
+    message("Something went wrong.")
+  }else{
+    res <- httr::content(r, as = "text")
+    df <- utils::read.csv(textConnection(res))
```
Comment (cgpu) [inline R/extract_participants.R:32]: `res`, `df` more descriptive names would be nice but I won't block the merge for that. Make a note to change those whenever you can.

File: R/list_cohorts.R [hunk 4]
```diff
@@ -29,15 +29,15 @@ list_cohorts <- function(baseurl,
                  query = list("teamId" = teamid,
                               "pageNumber" = page_number,
                               "pageSize" = page_size))
-  res <- httr::content(r)
-  if(length(res) == 0){
+  if (!r$status_code == 200) {
     message("No cohorts found. Or not able to connect with server.")
```
Comment (cgpu) [inline R/list_cohorts.R:33]: Here why `message` and not `error` to exit?

File: docs/404.html [hunk 5]
```diff
@@ -18,8 +18,8 @@
 <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>
```
Comment (cgpu) [inline docs/404.html:18]: Not sure if we need to commit or ignore the html here for pkgdown, can you verify?

Comment (cgpu) [reply docs/404.html:18]: I changed the branch of gh pages to new-api to check, all looks good https://lifebit-ai.github.io/cloudos/.
Feel free to ignore, I remembered how it works with Netlify and ignore of docs.

File: R/list_cohorts.R (hunk 4 above)
Comment (sk-sahu) [reply R/list_cohorts.R:33]: Good suggestion. There is `stop()` for error message and exit. I'll include that.

File: R/extract_participants.R (hunk 3 above)
Comment (sk-sahu) [reply R/extract_participants.R:32]: @cgpu for now these seems okay as long as they are not clash with R inbuilt objects/functions. I have seen some packages with such abbreviated representative names.

File: R/extract_genotypic_data.R (hunk 2 above)
Comment (sk-sahu) [reply R/extract_genotypic_data.R]: Done

**Comment by sk-sahu** [comment]
@cgpu I have tried to not build the vignettes `--no-build-vignettes` for R-CMD-Check as it require api-key. But this doesn't seems working. Any idea?

This is strange. Last time it worked in this -  https://github.com/lifebit-ai/cloudos/pull/5/commits/168682f80321a9081b0cd9f29e417a8f9bd09dab commit. Not sure how.

**Comment by cgpu** [comment]
> This is strange. Last time it worked in this - [168682f](https://github.com/lifebit-ai/cloudos/commit/168682f80321a9081b0cd9f29e417a8f9bd09dab) commit. Not sure how.

@sk-sahu i don't see the line with the `--no-build-vignettes` in https://github.com/lifebit-ai/cloudos/commit/168682f80321a9081b0cd9f29e417a8f9bd09dab.

Can you point the relevant line by checking in Browse files?

**Comment by sk-sahu** [comment]
> @sk-sahu i don't see the line with the --no-build-vignettes in 168682f.
> Can you point the relevant line by checking in Browse files?

@cgpu That's the point it didn't have `--no-build-vignettes` in this 168682f commit. But somehow it didn't render. But from that point onward it started this error in next commit.

File: check/cloudos.Rcheck/cloudos-Ex.Rout [hunk 6]
```diff
@@ -0,0 +1,147 @@
+
+R version 4.0.2 (2020-06-22) -- "Taking Off Again"
```
Comment (cgpu) [inline check/cloudos.Rcheck/cloudos-Ex.Rout:2]: do we need this files under version control @sk-sahu ?

Comment (sk-sahu) [reply check/cloudos.Rcheck/cloudos-Ex.Rout:2]: @cgpu No we don't.

This `check` directory created while local check and build. I need to delete this. Thank you for pointing out :)

Comment (sk-sahu) [reply check/cloudos.Rcheck/cloudos-Ex.Rout:2]: I'll add this to `.gitignore`

File: R/create_cohort.R [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+                                                 "accept" = "application/json, text/plain, */*",
+                                                 "content-type" = "application/json;charset=UTF-8")),
+                  query = list("teamId" = team_id),
+                  body = list("name" = cohort_name,
+                              "description" = cohort_desc,
+                              "moreFilters" = filters),
+                  encode = "json"
+  )
+  if (!r$status_code == 200) {
+    stop("Something went wrong. Not able to create a cohort")
```
Comment (cgpu) [inline R/create_cohort.R:38]: nice, I see `message` replaced for `stop`

File: R/create_cohort.R [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+  }else{
```
Comment (cgpu) [inline R/create_cohort.R:39]: `if (!r$status_code == 200)`

on this, I generally think `if` is more explicit and helps with matching assumptions with tests.

This is also jenny Bryan approved, see [here](https://speakerdeck.com/jennybc/code-smells-and-feels?slide=45)

**Review by cgpu — APPROVED** [review]
lgtm! 👍 

Only a minor note on (not) using `else`, see here [#r473191206](https://github.com/lifebit-ai/cloudos/pull/5/files#r473191206), but approving.

File: R/create_cohort.R (hunk 8 above)
Comment (sk-sahu) [reply R/create_cohort.R:39]: That's a legit point :+1:  This will make the code even more readable as well. Let me do that.

Comment (cgpu) [reply R/create_cohort.R]: @sk-sahu thanks for being on the Jenny Bryan side of life 😄 !

One note, after checking the code:
https://github.com/lifebit-ai/cloudos/blob/c7d081e2708f30af4eb91703d920abe3bc1c794f/R/create_cohort.R#L36-L43

Once you remove else, the idea is to replace it with the explicit `if statement`, not eliminate it altogether.
This is fine for now, I will open an issue and link it to your currently open PR to implement this in all functions.

So in principle as an example the following snippet, when `if`-ified would lokk like this:

https://github.com/lifebit-ai/cloudos/blob/1e1ec0098ff91108a6c1c8afacdf5a0f1d35a492/R/create_cohort.R#L37-L46

```r
if (!r$status_code == 200) { 
    stop("Something went wrong. Not able to create a cohort") 
 }
if (r$status_code == 200){
    message("Cohort named ", cohort_name, " created successfully. Bellow are the details") 
    res <- httr::content(r) 
    # into a dataframe 
    res_df <- do.call(rbind, res) 
    colnames(res_df) <- "details" 
    return(res_df) 
 } 
```

**Pull request merged**